public/app/features/plugins/admin/components/PluginDetailsHeaderDependencies.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { css } from '@emotion/css';
+import { GrafanaTheme2 } from '@grafana/data';
+import { useStyles2, Icon } from '@grafana/ui';
+import { CatalogPlugin } from '../types';
+
+type Props = {
+  plugin: CatalogPlugin;
+  className?: string;
+};
+
+const PluginIconClassName: Record<string, string> = {
+  datasource: 'gicon gicon-datasources',
+  panel: 'icon-gf icon-gf-panel',
+  app: 'icon-gf icon-gf-apps',
+  page: 'icon-gf icon-gf-endpoint-tiny',
+  dashboard: 'gicon gicon-dashboard',
+  default: 'icon-gf icon-gf-apps',
+};
+
+export function PluginDetailsHeaderDependencies({ plugin, className }: Props): React.ReactElement | null {
+  const styles = useStyles2(getStyles);
+  const pluginDependencies = plugin.details?.pluginDependencies;
+  const grafanaDependency = plugin.details?.grafanaDependency;
+  const hasNoDependencyInfo = !grafanaDependency && (!pluginDependencies || !pluginDependencies.length);
+
+  if (hasNoDependencyInfo) {
+    return null;
+  }
+
+  return (
+    <div className={className}>
+      <div className={styles.textBold}>Dependencies:</div>
+
+      {/* Grafana dependency */}
+      {Boolean(grafanaDependency) && (
+        <div>
+          <Icon name="grafana" />
+          Grafana {grafanaDependency}
+        </div>
+      )}
+
+      {/* Plugin dependencies */}
+      {pluginDependencies && pluginDependencies.length > 0 && (
```

Comment:
Any reason you added a test for the grafana dependency but not the plugin deps? Is it related to [this issue](https://github.com/grafana/grafana/issues/39063) somehow?

Reply:
Yes exactly, I was starting to add the test for them, but it was a bit hard to create the mocks for these scenarios with the setup that was serving us well until now. In the end I decided to move the update of the mocking (and with that adding tests for these) to a separate task (the issue that you linked). Wdyt?